### PR TITLE
added support for terraform

### PR DESCRIPTION
**Pull request by suren-khatana**
Added support for Terraform, now the bash script `deploy-idsvr-aws-eks.sh` provides an option to use either EKSCTL or TERRAFORM to provision AWS Infra and deploy Curity Identity Server.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,4 +9,16 @@ certs/*.srl
...
+templates
+.terraform
+.terraform.lock.hcl
+terraform.tfstate
+terraform-config/.terraform.tfstate.lock.info
+terraform-config/terraform.tfstate.backup
+terraform-config/1.4-vpc-creation.auto.tfvars
+terraform-config/2.4-eks-cluster-deployment.auto.tfvars
+.gitignore
+.gitignore
```
Comment (ju-cu) [inline .gitignore]: Why remove `.gitignore` from version control?
I assume git should ignore any `*.tfvars` file as well as `tfstate` files. 

Suggest to use the following entries in `.gitignore`:
```
# Local folder
.terraform/

# Dependency file
.terraform.lock.hcl

# Compiled files
*.tfstate
*.tfstate.backup
*.tfstate.lock.info

# Variable files
*.tfvars
```

**Comment by ju-cu** [comment]
Update README.md and add terraform cli to the list of prerequisites.

File: deploy-idsvr-aws-eks.sh [hunk 2]
```diff
@@ -27,14 +27,13 @@ greeting_message() {
...
-  echo -e "\n"
 }
 
 
 pre_requisites_check() {
   # Check if aws cli, kubectl, eksctl, helm & jq are installed
-  if ! [[ $(aws --version) && $(helm version) && $(jq --version) && $(eksctl version) ]]; then
-      echo "Please install aws cli, kubectl, eksctl, helm & jq to continue with the deployment .."
+  if ! [[ $(aws --version) && $(helm version) && $(jq --version) && $(eksctl version) && $(terraform version) ]]; then
+      echo "Please install aws cli, kubectl, eksctl, terraform, helm & jq to continue with the deployment .."
```
Comment (ju-cu) [inline deploy-idsvr-aws-eks.sh]: should say `eksctl or terraform` as the user can choose which one to use.

**Comment by ju-cu** [comment]
The `idsvr-config/helm-values.yaml.template` uses the configuration parameters `config.configurationSecret` and `config.configurationSecretItemName` which are both deprecated. Use `config.configuration` instead. (see https://github.com/curityio/idsvr-helm/tree/master/idsvr#configuration) .

**Comment by ju-cu** [comment]
`idsvr-config/idsvr-cluster-config.xml` contains a cluster key. Remove the cluster key as helm takes care of that. See https://github.com/curityio/idsvr-helm/blob/master/idsvr/templates/cluster-conf.yaml

Also, the PR renamed `simple-echo-api` to `example-api` on various places, but the `idsvr-config/idsvr-cluster-config.xml` still contains a client called `simple-echo-api`. Update the `idsvr-cluster-config.xml` and the related `example-api-config/api.js` to reflect the renaming properly.

File: .gitignore (hunk 1 above)
Comment (suren-khatana) [reply .gitignore]: Fixed.

File: deploy-idsvr-aws-eks.sh (hunk 2 above)
Comment (suren-khatana) [reply deploy-idsvr-aws-eks.sh]: Good catch ! , Fixed

File: README.md [hunk 3]
```diff
@@ -39,7 +40,7 @@ All of the services are running privately in the aws elastic kubernetes cluster
 
  2. Configuration
  
-    Cluster options could be configured by modifying `cluster-config/eks-cluster-config.json` file. Please ensure that the region set in the `cluster-config/eks-cluster-config.json` must match the region set for aws cli (~/.aws/config).
+    Cluster options could be configured by modifying `infrastructure-config/infra-config.json` file. Please ensure that the region set in the `infrastructure-config/infra-config.json` must match the region set for aws cli (~/.aws/config).
```
Comment (ju-cu) [inline README.md]: Grammar: either remove "must" or replace "ensure" with "note": 
a) Please ensure that ... matches ...
b) Please note that ... must match ...

File: deploy-idsvr-aws-eks.sh [hunk 4]
```diff
@@ -6,35 +6,35 @@ display_help() {
...
-    echo " --stop      shuts down the environment                                                                       "
-    echo " --delete    deletes the eks k8s cluster & identity server deployment                                         "
+    echo " --help                         shows this help message and exit                                                                 "
+    echo " --install                      creates a eks cluster & deploys curity identity server along with other components               "
+    echo " --start                        starts up the environment                                                                        "
+    echo " --stop                         shuts down the environment                                                                       "
+    echo " --load-balancer-public-ip      prints the public IP address of the load balancer                                                "
+    echo " --delete                       deletes the eks k8s cluster & identity server deployment                                         "
```
Comment (ju-cu) [inline deploy-idsvr-aws-eks.sh:14]: ... cluster and curity identity server deployment

File: example-api-config/example-api-ingress-nginx.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+kind: Ingress
+metadata:
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    nginx.ingress.kubernetes.io/configuration-snippet: |
+      phantom_token on;
+      phantom_token_client_credential api-gateway-client Password123;
+      phantom_token_introspection_endpoint curity;
+      phantom_token_scopes read;
+      error_log  /var/log/nginx/error.log  debug;
```
Comment (ju-cu) [inline example-api-config/example-api-ingress-nginx.yaml]: Having debug is very useful for developing but requires a debug-build of nginx, if I'm not mistaken. I suggest not to include a debug build or debug logs in this example but use a different log level as this is more in line with best practices.

File: infrastructure-config/infra-config.json [hunk 6]
```diff
@@ -37,30 +61,36 @@
...
-      "Type": "String",
       "Value": "4",
       "Description": "maximum size of the nodegroup"
     }
   ],
   "idsvr-config": [
+    {
+      "Name": "idsvr_namespace",
+      "Value": "curity",
+      "Description": "Name of the k8s namespace to deploy curity identity server"
```
Comment (ju-cu) [inline infrastructure-config/infra-config.json]: curity identity server => the Curity Identity Server

**Review by ju-cu — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (suren-khatana) [reply README.md]: Fixed

File: deploy-idsvr-aws-eks.sh (hunk 4 above)
Comment (suren-khatana) [reply deploy-idsvr-aws-eks.sh:14]: Fixed

File: example-api-config/example-api-ingress-nginx.yaml (hunk 5 above)
Comment (suren-khatana) [reply example-api-config/example-api-ingress-nginx.yaml]: Fixed

File: infrastructure-config/infra-config.json (hunk 6 above)
Comment (suren-khatana) [reply infrastructure-config/infra-config.json]: Fixed

**Pull request merged**